library/alloc/src/collections/binary_heap/mod.rs

```diff
@@ -1628,20 +1708,20 @@ impl<T: Ord> Iterator for DrainSorted<'_, T> {
 }
 
 #[unstable(feature = "binary_heap_drain_sorted", issue = "59278")]
-impl<T: Ord> ExactSizeIterator for DrainSorted<'_, T> {}
+impl<'a, T: Ord, A: Allocator + 'a> ExactSizeIterator for DrainSorted<'a, T, A> {}
 
 #[unstable(feature = "binary_heap_drain_sorted", issue = "59278")]
-impl<T: Ord> FusedIterator for DrainSorted<'_, T> {}
+impl<'a, T: Ord, A: Allocator + 'a> FusedIterator for DrainSorted<'a, T, A> {}
 
 #[unstable(feature = "trusted_len", issue = "37572")]
-unsafe impl<T: Ord> TrustedLen for DrainSorted<'_, T> {}
+unsafe impl<'a, T: Ord, A: Allocator + 'a> TrustedLen for DrainSorted<'a, T, A> {}
 
 #[stable(feature = "binary_heap_extras_15", since = "1.5.0")]
-impl<T: Ord> From<Vec<T>> for BinaryHeap<T> {
+impl<T: Ord, A: Allocator> From<Vec<T, A>> for BinaryHeap<T, A> {
```

Comment:
You can update the other direction too (on Vec).

And what about the one below? Can that be extended or are there inference failures?

Reply:
> You can update the other direction too (on Vec).

Isn't it updated? Line 1748?

> And what about the one below? Can that be extended or are there inference failures?

`impl<T: Ord, const N: usize> From<[T; N]> for BinaryHeap<T>`? AFAIK that can't work, because there's no way to instantiate the allocator.

Reply:
> Isn't it updated? Line 1748?

My bad.

> AFAIK that can't work, because there's no way to instantiate the allocator.

We could maybe do `where A: Allocator + Default` but there's no precedent for that so maybe that's best discussed in a separate PR.